### ISSUE TITLE
Manual Backup Failing

### ISSUE THREAD

**Issue by doodlebug70** [grokability/snipe-it#8080]
#### Please confirm you have done the following before posting your bug report:

- [x] I have enabled debug mode 
- [x] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

**Describe the bug**

Manual Backup of snipe-it failining due to permission errors. In advance apologies as this will be PEBKAC

**To Reproduce**
1. log in ad admin
2. go to settings > backups
3. click `generate backup`

**Expected behavior**

Manual backups should be created on clicking of `generate backup` button

**Screenshots**

**Server (please complete the following information):**
 - Snipe-IT Version  **v4.9.2 - build 4352 (master)**
 - OS: **Debian 10**
 - Web Server: **Apache 2.4**
 - PHP Version  **7.3**

**Desktop (please complete the following information):**
 - OS: **Debian 10**
 - Browser **Firefox** 
 - Version **68.8.0**

**Error Messages**

`Error: Starting backup... Backup failed because: unlink(/var/www/snipeit/storage/app/backup-temp/temp/db-dumps/mysql-snipeit.sql): Permission denied. `

**Additional context**
- Is this a fresh install or an upgrade?  **fresh**
- What method you used to install Snipe-IT (install.sh, manual installation, docker, etc) **git**
- Include what you've done so far in the installation, and if you got any error messages along the way.
- Indicate whether or not you've manually edited any data directly in the database **no**

Folder permissions:
```
ls -RalF
.:
total 20
drwxrwxr-x 4 snipeit www-data 4096 Feb 19 16:26 ./
drwxrwxr-x 7 snipeit www-data 4096 Feb  6 13:49 ../
-rwxrwxr-x 1 snipeit www-data   23 Feb  6 11:11 .gitignore*
drwxrwxr-x 3 snipeit www-data 4096 May  1 13:13 backup-temp/
drwxrwxr-x 3 snipeit www-data 4096 Feb  6 11:11 backups/

./backup-temp:
total 12
drwxrwxr-x 3 snipeit www-data 4096 May  1 13:13 ./
drwxrwxr-x 4 snipeit www-data 4096 Feb 19 16:26 ../
drwxr-xr-x 3 snipeit www-data 4096 May  1 13:13 temp/

./backup-temp/temp:
total 16
drwxr-xr-x 3 snipeit www-data 4096 May  1 13:13 ./
drwxrwxr-x 3 snipeit www-data 4096 May  1 13:13 ../
drwxr-xr-x 2 snipeit www-data 4096 May  1 13:13 db-dumps/
-rw-r--r-- 1 snipeit www-data 2281 May  1 13:13 manifest.txt

./backup-temp/temp/db-dumps:
total 204
drwxr-xr-x 2 snipeit www-data   4096 May  1 13:13 ./
drwxr-xr-x 3 snipeit www-data   4096 May  1 13:13 ../
-rw-r--r-- 1 snipeit www-data 196785 May  1 13:13 mysql-snipeit.sql

./backups:
total 16
drwxrwxr-x 3 snipeit www-data 4096 Feb  6 11:11 ./
drwxrwxr-x 4 snipeit www-data 4096 Feb 19 16:26 ../
-rwxrwxr-x 1 snipeit www-data   26 Feb  6 11:11 .gitignore*
drwxrwxr-x 2 snipeit www-data 4096 Feb  6 11:11 env-backups/

./backups/env-backups:
total 12
drwxrwxr-x 2 snipeit www-data 4096 Feb  6 11:11 ./
drwxrwxr-x 3 snipeit www-data 4096 Feb  6 11:11 ../
-rwxrwxr-x 1 snipeit www-data   13 Feb  6 11:11 .gitignore*
```

/etc/group has the following entry

`www-data:x:33:snipeit`

**Comment by boring10** [comment]
I think it is an issue with your backup-temp/temp directory. Try doing `chmod -R 775` on your backup-temp directory to ensure that there is rwxrwxr-x.

I'm not certain but I believe that items inside your backup-temp can be deleted since I think that is the staging area for when the backup is being generated before being bundled as a zip in the backups directory. I think that this may be where the error is stemming from at the moment. The www-data is used when you are using the web server and it currently doesn't have write permission for the backup-temp/temp directory.

chmod -R 777 on the storage directory has been suggested but that is probably a little heavier than you should start with.
https://github.com/snipe/snipe-it/issues/4458

**Comment by doodlebug70** [comment]
If I do from cli  `php artisan snipeit:backup -vvv`
 
I get error

      [Symfony\Component\Debug\Exception\FatalThrowableError]  Class 'ZipArchive' not found

**Comment by boring10** [comment]
This sounds similar to https://github.com/snipe/snipe-it/issues/3793 and https://github.com/snipe/snipe-it/issues/3761 which links to https://stackoverflow.com/questions/3872555/fatal-error-class-ziparchive-not-found-in/3872578#3872578 and http://www.php.net/manual/en/book.zip.php.

Running `php -v` should give you the current version of PHP that you have installed. This will help you narrow down what you need to do to install the php-zip for your system or compile PHP with the zip configuration option.

If you installed with the system package manager you would want to refer to https://packages.debian.org/sid/php-zip.

<blockquote><img src="https://cdn.sstatic.net/Sites/stackoverflow/Img/apple-touch-icon@2.png?v=73d79a89bded" width="48" align="right"><div>Stack Overflow</div><div><strong><a href="https://stackoverflow.com/questions/3872555/fatal-error-class-ziparchive-not-found-in">Fatal error: Class 'ZipArchive' not found in</a></strong></div><div>I have a problem that I install 'Archive_Zip 0.1.1' on Linux server, but when I try to run the script to create the zip file it gives the fatal error
  Fatal error: Class ZipArchive not found in ....</div></blockquote>
<blockquote><img src="https://www.php.net/favicon.ico" width="48" align="right"><div><strong><a href="http://www.php.net/manual/en/book.zip.php">PHP: Zip - Manual</a></strong></div></blockquote>
<blockquote><div><strong><a href="https://packages.debian.org/sid/php-zip">Debian -- Details of package php-zip in sid</a></strong></div><div>Zip module for PHP [default]</div></blockquote>

**Comment by doodlebug70** [comment]
Well now I am even more confused :)

`php-zip` is now install.   Restarted everything for good measure.

`/var/www/snipeit/storage/app/backup-temp` and all sub dirs were `chmod 775`.   Ran the backup and it still failed.   I then ran  `chmod -R 777 backup-temp`

Re-running the backup and now it works.   The confusing part...  the backup process reverts the directories under `backup-temp` to 775.

Out of curiosity I changed `backup-temp` back to 775 which was failing previously and re-re-ran the backup and it too works :/

Thank you for the  assistance, it's always nice to see a community that is so welcoming to help even it it's not necessarily an issue with the product
